### PR TITLE
Test profile input validation (EXPOSUREAPP-8524)

### PR DESCRIPTION
**Pull request by marcauberer**
Adds validation / input constraints to the fields `phone number`, `email`, `date of birth` and `postal code`

- Phone number gets validated with the `PHONE` validator pattern (valid phone number required to save profile)
- Email gets validated with the `EMAIL_ADDRESS` validator pattern (valid email address required to save profile)
- Postal code gets limited to 5 chars as we do not have longer postal codes in Germany
- Date of birth now rejects data selections in the future

Furthermore, this pr adds formatting of the phone number text field for more intuitivity.

We had a discussion about input validation at the QM department last week and the status quo was, that input validation is not planned to implement from SAP side. I thought to make an open source contribution, to provide a solution in case the interest exists ...

---
Internal Tracking ID: [EXPOSUREAPP-8524](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8524)

### CODE REVIEW

**Review by d4rken — DISMISSED** [review]
lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt [hunk 1]
```diff
@@ -55,10 +59,41 @@ class RATProfileCreateFragment : Fragment(R.layout.rat_profile_create_fragment),
...
+                if (Patterns.EMAIL_ADDRESS.matcher(it.toString()).matches()) {
+                    viewModel.emailChanged(it.toString())
+                } else {
+                    viewModel.emailChanged("")
+                }
+            }
+            emailInputEdit.setOnFocusChangeListener { _, hasFocus ->
+                // Validate email
+                if (!hasFocus && !Patterns.EMAIL_ADDRESS.matcher(emailInputEdit.text.toString()).matches()) {
+                    emailInputLayout.error = "Invalid email address"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt:92]: should be localised ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt [hunk 2]
```diff
@@ -55,10 +59,41 @@ class RATProfileCreateFragment : Fragment(R.layout.rat_profile_create_fragment),
...
+                if (Patterns.PHONE.matcher(it.toString()).matches()) {
+                    viewModel.phoneChanged(it.toString())
+                } else {
+                    viewModel.phoneChanged("")
+                }
+            }
+            phoneInputEdit.setOnFocusChangeListener { _, hasFocus ->
+                // Validate phone number
+                if (!hasFocus && !Patterns.PHONE.matcher(phoneInputEdit.text.toString()).matches()) {
+                    phoneInputLayout.error = "Invalid phone number"
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt:73]: should be localised ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt:92]: @harambasicluka pinged UA in #3764

to not increase complexity of PRs, maybe merge this and do a follow up PR?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt:92]: Ok, should we add least `strings` keys ? Normally UA update or translate them , but don't create new ones

Comment (marcauberer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt:92]: No problem. Will add that real quick ...

**Review by NataliaLemmerth — APPROVED** [review]
UA approved.

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt (hunk 1 above)
Comment (marcauberer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/coronatest/rat/profile/create/RATProfileCreateFragment.kt]: Should be fine now

**Review by mtwalli — APPROVED** [review]

**Review by BMItr — APPROVED** [review]
lgtm

**Pull request merged**